app/upgrades/v5_0/upgrades.go

```diff
@@ -0,0 +1,171 @@
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/module"
+	bankkeeper "github.com/cosmos/cosmos-sdk/x/bank/keeper"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/mezo-org/mezod/app/upgrades"
+	"github.com/mezo-org/mezod/utils"
+	evmkeeper "github.com/mezo-org/mezod/x/evm/keeper"
+	evmtypes "github.com/mezo-org/mezod/x/evm/types"
+)
+
+func CreateUpgradeHandler(
+	mm *module.Manager,
+	configurator module.Configurator,
+	keepers *upgrades.Keepers,
+) upgradetypes.UpgradeHandler {
+	return func(
+		ctx context.Context,
+		_ upgradetypes.Plan,
+		fromVM module.VersionMap,
+	) (module.VersionMap, error) {
+		sdkCtx := sdk.UnwrapSDKContext(ctx)
+
+		sdkCtx.Logger().Info("running v5.0.0 upgrade handler")
+
+		err := updateMEZOPrecompileVersion(sdkCtx, keepers.EvmKeeper)
+		if err != nil {
+			return nil, fmt.Errorf("failed to update MEZO precompile version: %w", err)
+		}
+
+		err = migrateBTCFromLegacyTigrisContracts(sdkCtx, keepers)
+		if err != nil {
+			return nil, fmt.Errorf("failed to migrate BTC from legacy Tigris contracts: %w", err)
+		}
+
+		return mm.RunMigrations(ctx, configurator, fromVM)
+	}
+}
+
+func updateMEZOPrecompileVersion(ctx sdk.Context, evmKeeper *evmkeeper.Keeper) error {
+	params := evmKeeper.GetParams(ctx)
+
+	ctx.Logger().Info(
+		"begin MEZO precompile version update",
+		"precompilesVersions",
+		params.PrecompilesVersions,
+	)
+
+	mezoVersionInfoIndex := slices.IndexFunc(
+		params.PrecompilesVersions,
+		func(versionInfo *evmtypes.PrecompileVersionInfo) bool {
+			// Compare bytes just in case to avoid any potential issues
+			// with string comparison.
+			return bytes.Equal(
+				evmtypes.HexAddressToBytes(versionInfo.PrecompileAddress),
+				evmtypes.HexAddressToBytes(evmtypes.MEZOTokenPrecompileAddress),
+			)
+		},
+	)
+
+	// 2 is the value of evmtypes.MEZOTokenPrecompileLatestVersion at
+	// the time of this upgrade. We avoid using the constant directly
+	// as it will change in the future so the actual value of the version
+	// used during this upgrade would perish over time.
+	params.PrecompilesVersions[mezoVersionInfoIndex].Version = 2
+
+	err := evmKeeper.SetParams(ctx, params)
+	if err != nil {
+		return err
+	}
+
+	ctx.Logger().Info(
+		"MEZO precompile version updated",
+		"precompilesVersions",
+		evmKeeper.GetParams(ctx).PrecompilesVersions,
+	)
+
+	return nil
+}
+
+func migrateBTCFromLegacyTigrisContracts(
+	ctx sdk.Context,
+	keepers *upgrades.Keepers,
+) error {
+	ctx.Logger().Info("begin btc migration from legacy Tigris contracts")
+
+	var legacyVoter, legacyRewardsDistributor, legacyChainFeeBuffer common.Address
+	//nolint:gocritic
+	if utils.IsTestnet(ctx.ChainID()) {
+		legacyVoter = common.HexToAddress("0x72F8dd7F44fFa19E45955aa20A5486E8EB255738")
+		legacyRewardsDistributor = common.HexToAddress("0x10B0E7b3411F4A38ca2F6BB697aA28D607924729")
+		legacyChainFeeBuffer = common.HexToAddress("0xFb14491168f6BbfFFf9880E936C4D0189e8105dA")
```

Comment:
`legacyChainFeeBuffer` will receive the migrated funds, right?
Shouldn't it be called just `chainFeeBuffer` (without legacy prefix)